src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIModelMapper.ChatCompletion.cs

```diff
@@ -0,0 +1,616 @@
+                {
+                    result.AllowParallelToolCalls = allowParallelToolCalls;
+                }
+
+                if (additionalProperties.TryGetValue(nameof(result.TopLogProbabilityCount), out int topLogProbabilityCountInt))
+                {
+                    result.TopLogProbabilityCount = topLogProbabilityCountInt;
+                }
+
+                if (additionalProperties.TryGetValue(nameof(result.Metadata), out IDictionary<string, string>? metadata))
+                {
+                    foreach (KeyValuePair<string, string> kvp in metadata)
+                    {
+                        result.Metadata[kvp.Key] = kvp.Value;
+                    }
+                }
+
+                if (additionalProperties.TryGetValue(nameof(result.StoredOutputEnabled), out bool storeOutputEnabled))
+                {
+                    result.StoredOutputEnabled = storeOutputEnabled;
+                }
+            }
+
+            if (options.Tools is { Count: > 0 } tools)
+            {
+                foreach (AITool tool in tools)
+                {
+                    if (tool is AIFunction af)
+                    {
+                        result.Tools.Add(ToOpenAIChatTool(af));
+                    }
+                }
+
+                switch (options.ToolMode)
+                {
+                    case AutoChatToolMode:
+                        result.ToolChoice = ChatToolChoice.CreateAutoChoice();
+                        break;
+
+                    case RequiredChatToolMode required:
+                        result.ToolChoice = required.RequiredFunctionName is null ?
+                            ChatToolChoice.CreateRequiredChoice() :
+                            ChatToolChoice.CreateFunctionChoice(required.RequiredFunctionName);
+                        break;
+                }
+            }
+
+            if (options.ResponseFormat is ChatResponseFormatText)
+            {
+                result.ResponseFormat = OpenAI.Chat.ChatResponseFormat.CreateTextFormat();
+            }
+            else if (options.ResponseFormat is ChatResponseFormatJson jsonFormat)
+            {
+                result.ResponseFormat = jsonFormat.Schema is { } jsonSchema ?
+                    OpenAI.Chat.ChatResponseFormat.CreateJsonSchemaFormat(
+                        jsonFormat.SchemaName ?? "json_schema",
+                        BinaryData.FromBytes(
+                            JsonSerializer.SerializeToUtf8Bytes(jsonSchema, OpenAIJsonContext.Default.JsonElement)),
+                        jsonFormat.SchemaDescription) :
+                    OpenAI.Chat.ChatResponseFormat.CreateJsonObjectFormat();
+            }
+        }
+
+        return result;
+    }
+
+    private static AITool FromOpenAIChatTool(ChatTool chatTool)
+    {
+        Dictionary<string, object?> additionalProperties = new();
```

Comment:
Should we use AdditionalPropertiesDictionary here? It's not required, but it would provide consistency, including using OrdinalIgnoreCase.